app/src/main/java/org/xbmc/kore/ui/generic/SubtitlesOptionsDialog.java

```diff
@@ -0,0 +1,313 @@
+package org.xbmc.kore.ui.generic;
+
+
+import android.app.Activity;
+
+import android.os.Handler;
+import android.support.v4.app.FragmentActivity;
+import android.text.TextUtils;
+import android.widget.Toast;
+
+import org.xbmc.kore.R;
+import org.xbmc.kore.host.HostConnectionObserver;
+import org.xbmc.kore.host.HostInfo;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.ApiCallback;
+import org.xbmc.kore.jsonrpc.ApiMethod;
+import org.xbmc.kore.jsonrpc.HostConnection;
+import org.xbmc.kore.jsonrpc.method.Addons;
+import org.xbmc.kore.jsonrpc.method.GUI;
+import org.xbmc.kore.jsonrpc.method.Input;
+import org.xbmc.kore.jsonrpc.method.Player;
+import org.xbmc.kore.jsonrpc.type.ListType;
+import org.xbmc.kore.jsonrpc.type.PlayerType;
+import org.xbmc.kore.utils.LogUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Dialog that presents a list of options related to subtitles.
+ */
+public class SubtitlesOptionsDialog
+        implements HostConnectionObserver.PlayerEventsObserver,
+                    GenericSelectDialog.GenericSelectDialogListener{
+
+
+    // An interface to be implemented by everyone interested in RemoteRequired events
+    public interface SubtitlesOptionsDialogListener {
+        void onRemoteRequired();
+    }
+
+    private List<SubtitlesOptionsDialogListener> listeners = new ArrayList<SubtitlesOptionsDialogListener>();
+
+    private static final String TAG = LogUtils.makeLogTag(SubtitlesOptionsDialog.class);
+
+    /**
+     * List of available subtitles
+     */
+    private List<PlayerType.Subtitle> availableSubtitles;
+    private int currentSubtitleIndex = -1;
+
+    // Number of explicitly added options for subtitles (to subtract from the
+    // number of subtitles returned by Kodi)
+    static final int ADDED_SUBTITLE_OPTIONS = 3;
+
+    /**
+     * Constants for the general select dialog
+     */
+    private final static int SELECT_SUBTITLES = -111;
+
+    /**
+     * Host manager from which to get info about the current XBMC
+     */
+    private HostManager hostManager;
+
+    /**
+     * Handler on which to post RPC callbacks
+     */
+    private Handler callbackHandler = new Handler();
+
+    /**
+     * Activity to communicate potential actions that change what's playing
+     */
+    private HostConnectionObserver hostConnectionObserver;
+
+    /**
+     * The current active player id
+     */
+    private int currentActivePlayerId = -1;
+
+    /**
+     * Default callback for methods that don't return anything
+     */
+    private ApiCallback<String> defaultStringActionCallback = ApiMethod.getDefaultActionCallback();
+
+
+
+    private FragmentActivity dialogActivity;
```

Comment:
I find `dialogActivity` confusing. Why not name it `fragmentActivity` or just `activity`?